### PR TITLE
VxAdmin: Update tabulation logic (manual tallies and reports) to use ballot style group

### PR DESCRIPTION
**Pull request by carolinemodic**
## Overview
https://github.com/votingworks/vxsuite/issues/5183
In adding multi-language ballot styles as "sub" ballot styles in our election format we made the ballot style filters/groups for reporting in VxAdmin too specific. To protect privacy insitead of reporting on ballot styles athat are language specific like 1-Ma_en we should instead report on the aggregated grouping 1-Ma. 

In an effort to make it as clear as possible what is happening where in the codebase I did a pretty large sweeping renaming changes to hopefully always refer to it as ballotStyleGroupId when referencing the parent of a group and ballotStyleId refers to the specific sub-language style id. 

Manual tallies will now be entered on a "group" level as well. 

## Demo Video or Screenshot
Ballot styles replaced with language - agnostic ids through VxAdmin, screenshots not particularly illustrative for this change. 

## Testing Plan
Manually tested with a multi-language primary election getting reports in all formats (pdf, csv, err) for full tally report, and available formats (pdf, csv) for filtered and grouped tally reports with filters and groups by ballot style, party, etc. Manually tested basic WIA flow. Manually tested adding manual tallies and verifying tabulation results are correct with them. Etc. Tested both with generated cvrs and manually created ones with cvrs across different languages for the same grouped ballot style. 

## Checklist

- [x] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [x] I have added a screenshot and/or video to this PR to demo the change
- [x] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

**Comment by carolinemodic** [comment]
cc @adghayes

File: libs/types/src/tabulation.ts [hunk 1]
```diff
@@ -76,7 +76,7 @@ export interface Filter {
  * Attributes that always exist for every cast vote record.
  */
 export interface CastVoteRecordAttributes {
-  readonly ballotStyleGroupId: BallotStyleGroupId;
+  readonly ballotStyleGroupId: Id;
```
Comment (jonahkagan) [inline libs/types/src/tabulation.ts:79]: Why this change?

File: libs/types/src/election.ts [hunk 2]
```diff
@@ -308,15 +308,26 @@ export const PrecinctsSchema = z
...
 }
+
+export type BallotStyleGroupId = NewType<string, 'BallotStyleGroupId'>;
+export interface ParentBallotStyle {
+  readonly id: BallotStyleGroupId;
+  readonly precincts: readonly PrecinctId[];
+  readonly districts: readonly DistrictId[];
+  readonly partyId?: PartyId;
+  readonly languages?: readonly LanguageCode[]; // TODO(kofi): Make required.
+}
```
Comment (jonahkagan) [inline libs/types/src/election.ts:329]: I notice that in some places we use default language ballot styles and in some places we use `ParentBallotStyle`. I think having a mix of both concepts makes things a bit confusing. Can we pick one?

Also, if we go with parent ballot styles, maybe we could call it `BallotStyleGroup` instead and remove the languages field (keeping only the fields that are common across all of the ballot styles in the group).

**Review by jonahkagan — COMMENTED** [review]
Thanks for separating out the different commits. Very helpful!

(Side note: there are now so many files that I can only view the PR commit by commit, Github freezes when I try to view the full diff)

File: libs/types/src/tabulation.ts (hunk 1 above)
Comment (carolinemodic) [reply libs/types/src/tabulation.ts:79]: I forget exactly but I think there were insurmountable other issues if this isn't a string as all other fields here are strings and that is utilized in other parts of the code base that generalize the keys

**Review by jonahkagan — APPROVED** [review]
💥 :shipit:
